# Netflix Movies Data Analysis

## Project Introduction

This project presents a comprehensive exploratory data analysis (EDA) of movies available on Netflix. The dataset contains valuable information about various movies, including their titles, genres, release years, popularity scores, vote counts and other relevant attributes.

### Objective
The main goal of this analysis is to uncover meaningful insights from the Netflix movie dataset that can help understand trends in movie production, genre popularity and audience engagement. These insights will support **informed business decisions** in content strategy and recommendation systems.

### Key Business Questions
This analysis aims to answer the following important questions:

1. What is the most frequent genre of movies released on Netflix?  
2. Which genre has the highest number of votes?  
3. What movie got the highest popularity? What is its genre?  
4. What movie got the lowest popularity? What is its genre?  
5. Which year has the most filmed movies?

### Approach
We will perform data cleaning, type conversion, exploratory data analysis and visualization to extract actionable insights from the dataset.


In [24]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [46]:
df = pd.read_csv("mymoviedb.csv",lineterminator='\n')

In [47]:
df.head()

,Release_Date,Title,Overview,Popularity,Vote_Count,Vote_Average,Original_Language,Genre,Poster_Url
0,2021-12-15,Spider-Man: No Way Home,Peter Parker is unmasked and no longer able to...,5083.954,8940,8.3,en,"Action, Adventure, Science Fiction",https://image.tmdb.org/t/p/original/1g0dhYtq4i...
1,2022-03-01,The Batman,"In his second year of fighting crime, Batman u...",3827.658,1151,8.1,en,"Crime, Mystery, Thriller",https://image.tmdb.org/t/p/original/74xTEgt7R3...
2,2022-02-25,No Exit,Stranded at a rest stop in the mountains durin...,2618.087,122,6.3,en,Thriller,https://image.tmdb.org/t/p/original/vDHsLnOWKl...
3,2021-11-24,Encanto,"The tale of an extraordinary family, the Madri...",2402.201,5076,7.7,en,"Animation, Comedy, Family, Fantasy",https://image.tmdb.org/t/p/original/4j0PNHkMr5...
4,2021-12-22,The King's Man,As a collection of history's worst tyrants and...,1895.511,1793,7.0,en,"Action, Adventure, Thriller, War",https://image.tmdb.org/t/p/original/aq4Pwv5Xeu...


In [48]:
df.sample(5)

,Release_Date,Title,Overview,Popularity,Vote_Count,Vote_Average,Original_Language,Genre,Poster_Url
6048,2004-03-17,Spartacus,Sentenced to spend out the rest of his adult l...,18.425,52,6.8,en,"Action, Drama",https://image.tmdb.org/t/p/original/4qDIb9K9qL...
9358,2011-10-12,The Artist,"Hollywood, 1927: As silent movie star George V...",13.778,2651,7.4,en,"Drama, Comedy, Romance",https://image.tmdb.org/t/p/original/khtVL4abxY...
8570,2015-05-01,Flying Colors,"A high school girl, Sayaka Kudō was the bottom...",14.552,77,7.4,ja,"Comedy, Drama",https://image.tmdb.org/t/p/original/m53V2l1A5i...
4474,2020-11-26,Happiest Season,A young woman's plans to propose to her girlfr...,22.690,783,7.4,en,"Romance, Comedy",https://image.tmdb.org/t/p/original/vzec9kkOSE...
9234,1985-05-15,Mishima: A Life in Four Chapters,A fictional account of the life of Japanese au...,13.917,179,7.8,en,Drama,https://image.tmdb.org/t/p/original/v8U0mYf77r...


In [49]:
df.shape

(9827, 9)

In [50]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9827 entries, 0 to 9826
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Release_Date       9827 non-null   object 
 1   Title              9827 non-null   object 
 2   Overview           9827 non-null   object 
 3   Popularity         9827 non-null   float64
 4   Vote_Count         9827 non-null   int64  
 5   Vote_Average       9827 non-null   float64
 6   Original_Language  9827 non-null   object 
 7   Genre              9827 non-null   object 
 8   Poster_Url         9827 non-null   object 
dtypes: float64(2), int64(1), object(6)
memory usage: 691.1+ KB


In [51]:
df['Genre'].head()

0    Action, Adventure, Science Fiction
1              Crime, Mystery, Thriller
2                              Thriller
3    Animation, Comedy, Family, Fantasy
4      Action, Adventure, Thriller, War
Name: Genre, dtype: object

In [52]:
df.duplicated().sum()

np.int64(0)

In [68]:
df.isna().sum()

Release_Date      0
Title             0
Popularity        0
Vote_Count        0
Vote_Average    100
Genre             0
dtype: int64

In [53]:
df.describe()

,Popularity,Vote_Count,Vote_Average
count,9827.000000,9827.000000,9827.000000
mean,40.326088,1392.805536,6.439534
std,108.873998,2611.206907,1.129759
min,13.354000,0.000000,0.000000
25%,16.128500,146.000000,5.900000
50%,21.199000,444.000000,6.500000
75%,35.191500,1376.000000,7.100000
max,5083.954000,31077.000000,10.000000


### Exploration Summary: for Data Preprocessing

* we have dataset with `9827` rows and `9` columns.
* Dataset has `Duplicated` values
* Convert the `date` column from `object` data type to `datetime` for proper temporal analysis and operations.
* in dataset like, `Overview`,`Original_language` and `Postar_url` These columns are not relevant to the business questions being analyzed.
* there are noticeable can outlier in `Popularity` column.
* `Vote_Avarage` better be categorised for proper analysis.
* `Genre` Column has **Coma** Separated values with white space so we need to handle it.
* Remove `NaN` values in `Vote_Avarage` column.

In [54]:
df['Release_Date'].dtype

dtype('O')

### Convert Data type to datetime

In [55]:
df['Release_Date']=pd.to_datetime(df['Release_Date'])

print(df['Release_Date'].dtype)

datetime64[ns]


### Only Need year for analaysis, so keep only years

In [56]:
df['Release_Date'] = df['Release_Date'].dt.year

df['Release_Date'].dtypes

dtype('int32')

In [57]:
df.head()

,Release_Date,Title,Overview,Popularity,Vote_Count,Vote_Average,Original_Language,Genre,Poster_Url
0,2021,Spider-Man: No Way Home,Peter Parker is unmasked and no longer able to...,5083.954,8940,8.3,en,"Action, Adventure, Science Fiction",https://image.tmdb.org/t/p/original/1g0dhYtq4i...
1,2022,The Batman,"In his second year of fighting crime, Batman u...",3827.658,1151,8.1,en,"Crime, Mystery, Thriller",https://image.tmdb.org/t/p/original/74xTEgt7R3...
2,2022,No Exit,Stranded at a rest stop in the mountains durin...,2618.087,122,6.3,en,Thriller,https://image.tmdb.org/t/p/original/vDHsLnOWKl...
3,2021,Encanto,"The tale of an extraordinary family, the Madri...",2402.201,5076,7.7,en,"Animation, Comedy, Family, Fantasy",https://image.tmdb.org/t/p/original/4j0PNHkMr5...
4,2021,The King's Man,As a collection of history's worst tyrants and...,1895.511,1793,7.0,en,"Action, Adventure, Thriller, War",https://image.tmdb.org/t/p/original/aq4Pwv5Xeu...


### Dropping the columns

In [58]:
cols=['Overview','Original_Language','Poster_Url']

In [59]:
df.drop(columns=cols,axis=1,inplace=True) # axis 1 means columns, 0 means row/index
df.columns

Index(['Release_Date', 'Title', 'Popularity', 'Vote_Count', 'Vote_Average',
       'Genre'],
      dtype='object')

In [60]:
df.head()

,Release_Date,Title,Popularity,Vote_Count,Vote_Average,Genre
0,2021,Spider-Man: No Way Home,5083.954,8940,8.3,"Action, Adventure, Science Fiction"
1,2022,The Batman,3827.658,1151,8.1,"Crime, Mystery, Thriller"
2,2022,No Exit,2618.087,122,6.3,Thriller
3,2021,Encanto,2402.201,5076,7.7,"Animation, Comedy, Family, Fantasy"
4,2021,The King's Man,1895.511,1793,7.0,"Action, Adventure, Thriller, War"


### **Categorised `Vote_Average` column**
* Here we cut the `Vote_Average` columns and set 4 category like: **popular**, **average**, **below_avearge**, **not_popular**

In [61]:
# User define function for categorised
def categorised_col(df, col, labels):

    edges=[df[col].describe()['min'],
          df[col].describe()['25%'],
          df[col].describe()['50%'],
          df[col].describe()['75%'],
          df[col].describe()['max']]
    
    df[col]=pd.cut(x=df[col],bins=edges,labels=labels,duplicates='drop')
    return df

In [62]:
labels =['not_popular','below_average','average','popular']

In [63]:
categorised_col(df,'Vote_Average',labels)

df['Vote_Average'].unique()

['popular', 'below_average', 'average', 'not_popular', NaN]
Categories (4, object): ['not_popular' < 'below_average' < 'average' < 'popular']

In [64]:
df.head()

,Release_Date,Title,Popularity,Vote_Count,Vote_Average,Genre
0,2021,Spider-Man: No Way Home,5083.954,8940,popular,"Action, Adventure, Science Fiction"
1,2022,The Batman,3827.658,1151,popular,"Crime, Mystery, Thriller"
2,2022,No Exit,2618.087,122,below_average,Thriller
3,2021,Encanto,2402.201,5076,popular,"Animation, Comedy, Family, Fantasy"
4,2021,The King's Man,1895.511,1793,average,"Action, Adventure, Thriller, War"


In [66]:
df['Vote_Average'].value_counts()

Vote_Average
not_popular      2467
popular          2450
average          2412
below_average    2398
Name: count, dtype: int64

In [67]:
df.isna().sum()

Release_Date      0
Title             0
Popularity        0
Vote_Count        0
Vote_Average    100
Genre             0
dtype: int64

### Here NaN value number is `100` which are too small compare to data shape, so drop those rows

In [69]:
df.dropna(inplace=True)

In [70]:
df.isna().sum()

Release_Date    0
Title           0
Popularity      0
Vote_Count      0
Vote_Average    0
Genre           0
dtype: int64

In [72]:
df.shape

(9727, 6)